### PR TITLE
Code parrot minor fixes/niceties

### PR DESCRIPTION
**Pull request by ncoop57**
# What does this PR do?

Hi there, really excited by HF expanding into Deep Learning for Software Engineering! This PR just adds a few fixes/niceties I found while testing out the scripts. Specifically:
1. Move the training code to a main function to allow using TPUs (TPU testing still needs to be done, but GPU training still works) with accelerate since it expects some entry function
2. Add some additional flags to human eval script for specifying which device to run the `text-generation` pipeline on and to specify how many tasks to evaluate against as it was previously hard-coded to 4
3. Update documentation to have the correct batch size flag for the validation set (was `eval_batch_size`, but should be `valid_batch_size`), add command for human eval script to use flag `--HF_ALLOW_CODE_EVAL="1"` since the program will throw an exception after generating all the samples, and add a note to make sure to install git-lfs for initializing/training the model.
4. Fix one of the requirements (`huggingface-hub`) to be `0.1.0` so that it does not conflict with the other libraries.

Since I couldn't get TPU training to work even with moving everything into the main function, I can separate out that commit if y'all would prefer into a separate WIP PR while I work on it. Just lemme know :nerd_face:!

You can verify my changes with this colab! https://colab.research.google.com/drive/1Tn0wsqqNbAEpxDn2AFs13Ssqwb0r3o-x?usp=sharing

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

- research_projects/codeparrot: @lvwerra

### CODE REVIEW

File: examples/research_projects/codeparrot/scripts/arguments.py [hunk 1]
```diff
@@ -83,6 +83,9 @@ class HumanEvalArguments:
         metadata={"help": "Model name or path of model to be evaluated."},
     )
     num_workers: Optional[int] = field(default=None, metadata={"help": "Number of workers used for code evaluation."})
+    num_tasks: Optional[int] = field(
+        default=None, metadata={"help": "The number of human-eval tasks to run. If not include, will default to running all tasks."}
```
Comment (lvwerra) [inline examples/research_projects/codeparrot/scripts/arguments.py:87]:
```suggestion
        default=None, metadata={"help": "The number of human-eval tasks to run. If not included all tasks are evaluated."}
```

File: examples/research_projects/codeparrot/README.md [hunk 2]
```diff
@@ -125,7 +127,8 @@ python scripts/human_eval.py --model_ckpt lvwerra/codeparrot \
 --do_sample True \
 --temperature 0.2 \
 --top_p 0.95 \
---n_samples=200
+--n_samples=200 \
+--HF_ALLOW_CODE_EVAL="1"
```
Comment (lvwerra) [inline examples/research_projects/codeparrot/README.md:131]: This was set to 0 by default on purpose, but I agree we can make the flag more visible. 

```suggestion
--HF_ALLOW_CODE_EVAL="0"
```

The reason I would like to avoid being able to run the code by default is that this metric executes the generated snippets on the machine. The generations could be harmful so it is important that the user is aware of the risks. But I agree that getting error only after the generation is complete is annoying as this can take a long time. We could do a quick test at the beginning e.g.:

```python
try:
    _ = metric.compute(references=[''], predictions=[['']])
except Exception as e:
    print("Code evaluation not enabled. Read the warning below carefully and then use `--HF_ALLOW_CODE_EVAL '1'` flag to enable code evaluation.")
    raise e
```

File: examples/research_projects/codeparrot/README.md [hunk 3]
```diff
@@ -75,6 +75,8 @@ python scripts/initialize_model.py \
 ```
 This will initialize a new model with the architecture and configuration of `gpt2-large` and use the tokenizer to appropriately size the input embeddings. Finally, the initilaized model is pushed the the hub.
 
+**Note** Make sure you have git-lfs installed. You can find instructions for how to install it [here](https://git-lfs.github.com/).
```
Comment (lvwerra) [inline examples/research_projects/codeparrot/README.md:78]: Good point but lets add this to the installation section.
```suggestion
```

**Review by lvwerra — COMMENTED** [review]
Thanks @ncoop57 for the fixes, looks good to me. I left a few minor comments.

Regarding the TPU support: I agree, let's do this in a separate PR and consider merging it once it is working.

File: examples/research_projects/codeparrot/README.md (hunk 2 above)
Comment (ncoop57) [reply examples/research_projects/codeparrot/README.md]: Makes total sense to give it as 0 so that they don't execute dangerous code by accident. I'll add this try except to the beginning so people don't spend their wheels!

**Comment by ncoop57** [comment]
Alrighty @lvwerra, I think this PR should be (:crossed_fingers:) good to go :rocket:! Lemme know if anything else needs to be done!

File: examples/research_projects/codeparrot/scripts/human_eval.py [hunk 4]
```diff
@@ -51,14 +51,23 @@ def main():
...
     code_eval_metric = load_metric("code_eval")
 
+    # Run a quick test to see if code evaluation is enabled
+    try:
+        _ = code_eval_metric.compute(references=[""], predictions=[[""]])
+    except Exception as e:
+        print(
+            'Code evaluation not enabled. Read the warning below carefully and then use `--HF_ALLOW_CODE_EVAL="1"` flag to enable code evaluation.'
+        )
+        raise e
```
Comment (lvwerra) [inline examples/research_projects/codeparrot/scripts/human_eval.py:67]: It would be good to be more concrete about the exception. If there is another issue (e.g. on Windows this will throw a `NotImplementedError`) the user would be very confused.

I should have thought a bit more about my previous suggestion!

```suggestion
    except ValueError as exception:
        print(
            'Code evaluation not enabled. Read the warning below carefully and then use `--HF_ALLOW_CODE_EVAL="1"` flag to enable code evaluation.'
        )
        raise exception
```

**Review by lvwerra — COMMENTED** [review]
Hi @ncoop57, thanks for integrating all the feedback. Just a minor comment and then I think it is good to go!

File: examples/research_projects/codeparrot/scripts/human_eval.py (hunk 4 above)
Comment (ncoop57) [reply examples/research_projects/codeparrot/scripts/human_eval.py:67]: Oooooo i didn't realize that was a thing, I'll need to remember this in future!

Comment (lvwerra) [reply examples/research_projects/codeparrot/scripts/human_eval.py:67]: I just noticed that you undid  the fix of `self.tokenizer` fix in the `ConstantLengthDataset`. Could you add it back?

Comment (ncoop57) [reply examples/research_projects/codeparrot/scripts/human_eval.py]: > I just noticed that you undid the fix of `self.tokenizer` in the `ConstantLengthDataset`. Could you add it back?

On it! Forgot about that too haha

**Review by lvwerra — APPROVED** [review]
Thanks for adding these tweaks! 🚀
